src/NATS.Client.Core/NatsTlsOpts.cs

```diff
@@ -45,47 +46,37 @@ public sealed record NatsTlsOpts
     /// <summary>
     /// String or file path to PEM-encoded X509 Certificate
     /// </summary>
-    /// <remarks>
-    /// You can use <see cref="NatsTlsOpts"/> to load the certificate from a file
-    /// or you can pass the PEM-encoded certificate directly as string. If the value
-    /// looks like a PEM-encoded pattern (for example if it starts with "-----BEGIN CERTIFICATE-----")
-    /// it would be used as PEM-encoded string otherwise it would be treated as file path.
-    /// </remarks>
     public string? CertFile { get; init; }
 
     /// <summary>
     /// String or file path to PEM-encoded Private Key
     /// </summary>
-    /// <remarks>
-    /// You can use <see cref="NatsTlsOpts"/> to load the private key from a file
-    /// or you can pass the PEM-encoded private key directly as string. If the value
-    /// looks like a PEM-encoded pattern (for example if it starts with "-----BEGIN PRIVATE KEY-----")
-    /// it would be used as PEM-encoded string otherwise it would be treated as file path.
-    /// </remarks>
     public string? KeyFile { get; init; }
 
+    /// Callback that loads Client Certificate
+    public Func<ValueTask<X509Certificate2>>? LoadClientCert { get; init; }
```

Comment:
Is it typical to use the `Async` suffix on a callback that returns a Task/ValueTask?  Should this be `LoadClientCertAsync`?

Reply:
Not sure about the naming. I wouldn't've thought so.

btw why do we need a callback? can it be just `X509Certificate2`?

Reply:
Callback is so we can support reloading certs - for example if they expire.  Right now `TlsCerts` is constructed on every initial connect in `NatsConnection.InitialConnectAsync`

Although this behavior should probably be documented.

Reply:
of course! makes sense.

Reply:
> Right now `TlsCerts` is constructed on every initial connect in `NatsConnection.InitialConnectAsync`

should we reload on reconnect? (not for this PR)